recipes/mpir/all/conanfile.py

```diff
@@ -101,6 +98,22 @@ def _build_visual_studio(self):
         for vcxproj_path in self._vcxproj_paths:
             msbuild.build(vcxproj_path, platforms=self._platforms, upgrade_project=False)
 
+    @contextlib.contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "apple-clang":
+            env_build = {"CC": tools.XCRun(self.settings).cc,
+                         "CXX": tools.XCRun(self.settings).cxx}
+            if hasattr(self, "settings_build"):
+                # there is no CFLAGS_FOR_BUILD/CXXFLAGS_FOR_BUILD
+                xcrun = tools.XCRun(self.settings_build)
+                flags = " -Wno-implicit-function-declaration -isysroot {} -arch {}".format(xcrun.sdk_path, tools.to_apple_arch(self.settings_build.arch))
+                env_build["CC_FOR_BUILD"] = xcrun.cc + flags
+                env_build["CXX_FOR_BUILD"] = xcrun.cxx + flags
```

Comment:
Wouldn't it be more elegant and robust to get `self._autotools.flags` and `self._autotools.cxxflags` in `_configure_autotools()`, then inject those values in `CC_FOR_BUILD` and `CXX_FOR_BUILD` if cross-building (I guess those values are required for some reason in this case)?